src/index.js

```diff
@@ -20,10 +20,185 @@
 
 import SpanContext from './span_context';
 import Span from './span';
+import ConstSampler from './samplers/const_sampler';
+import InMemoryReporter from './reporters/in_memory_reporter';
+import NoopReporter from './reporters/noop_reporter';
+import ProbabilisticSampler from './samplers/probabilistic_sampler';
+import RatelimitingSampler from './samplers/ratelimiting_sampler';
+import RemoteReporter from './reporters/remote_reporter';
+import RemoteSampler from './samplers/remote_sampler';
 import Tracer from './tracer';
+import UDPSender from './reporters/udp_sender';
+import {Validator} from 'jsonschema';
+
+
+let configSchema = {
+    'type': 'object',
+    'required': ['jaeger'],
+    'properties': {
+        'jaeger': {
+            'type': 'object',
+            '$ref': '/jaeger'
+        }
+    },
+};
+
+let constSamplerSchema = {
+    'id': '/const',
+    'type': 'object',
+    'properties': {
+        'type': {'type': 'string' },
+        'param': {'type': 'boolean' }
+    },
+    'required': ['type', 'param'],
+    'additionalProperties': false
+};
+
+let probabilisticSamplerSchema = {
+    'id': '/probabilistic',
+    'type': 'object',
+    'properties': {
+        'type': {'type': 'string' },
+        'param': {'type': 'number' }
+    },
+    'required': ['type', 'param'],
+    'additionalProperties': false
+};
+
+let ratelimitingSamplerSchema = {
+    'id': '/ratelimiting',
+    'type': 'object',
+    'properties': {
+        'type': {'type': 'string' },
+        'param': {'type': 'number' }
+    },
+    'required': ['type', 'param'],
+    'additionalProperties': false
+};
+
+let remoteSamplerSchema = {
+    'id': '/remote',
+    'type': 'object',
+    'properties': {
+        'type': {'type': 'string' },
+        'param': {'type': 'string' },
+        'managerHostPort': {'type': 'string'}
+    },
+    'required': ['type', 'param'],
+    'additionalProperties': false
+};
+
+let jaegerSchema = {
+    'id': '/jaeger',
+    'type': 'object',
+    'required': ['serviceName'],
```

Comment:
Having a brand new entry in the config provides no more guarantee that the service name will be correct vs. if we require them to pass it to initTracer as a literal string. But the latter allows us to build a wrapper in the internal repo that will look up the service name from env variable.

Also, our code should be able to function with a completely empty config, which by default should be completely empty in production, to use the internal tracer defaults. The philosophy of an empty config does not work well if there is a required `serviceName` field in that config.